### PR TITLE
Feature/effect spec helpers

### PR DESCRIPTION
**Pull request by VectorGamez**

### CODE REVIEW

File: Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h [hunk 1]
```diff
@@ -38,4 +39,13 @@ class GASEXTENSIONS_API UGASExtAbilitySystemFunctionLibrary final : public UBlue
 
     UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
     static bool IsGameplayEffectHandleValid( FActiveGameplayEffectHandle gameplay_effect_handle );
+
+    UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
+    static FGameplayEffectContextHandle GetContextHandle( const FGameplayEffectSpec & gameplay_effect_spec );
```
Comment (TheEmidee) [inline Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h:44]: I'd rename `GetContextHandleFromGameplayEffectSpec`

File: Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h [hunk 2]
```diff
@@ -38,4 +39,13 @@ class GASEXTENSIONS_API UGASExtAbilitySystemFunctionLibrary final : public UBlue
 
     UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
     static bool IsGameplayEffectHandleValid( FActiveGameplayEffectHandle gameplay_effect_handle );
+
+    UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
+    static FGameplayEffectContextHandle GetContextHandle( const FGameplayEffectSpec & gameplay_effect_spec );
+
+    UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
+    static FGameplayTagContainer GetTargetTagContainer( const FGameplayEffectSpec & gameplay_effect_spec );
```
Comment (TheEmidee) [inline Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h:47]: I'd rename `GetTargettagContainerFromGameplayEffectSpec`

File: Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h [hunk 3]
```diff
@@ -38,4 +39,13 @@ class GASEXTENSIONS_API UGASExtAbilitySystemFunctionLibrary final : public UBlue
...
     static bool IsGameplayEffectHandleValid( FActiveGameplayEffectHandle gameplay_effect_handle );
+
+    UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
+    static FGameplayEffectContextHandle GetContextHandle( const FGameplayEffectSpec & gameplay_effect_spec );
+
+    UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
+    static FGameplayTagContainer GetTargetTagContainer( const FGameplayEffectSpec & gameplay_effect_spec );
+
+    UFUNCTION( BlueprintPure, Category = "Ability|GameplayEffects" )
+    static float GetFloatAttribute( const FGameplayEffectSpec & gameplay_effect_spec, FGameplayAttribute attribute );
```
Comment (TheEmidee) [inline Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h:50]: I think you should just create a function that returns the effect causer. Getting a float attribute from it already exists

**Review by TheEmidee — CHANGES_REQUESTED** [review]

**Review by TheEmidee — APPROVED** [review]

File: Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h (hunk 3 above)
Comment (VectorGamez) [reply Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h]: We can also get the effect causer from the context already. That makes us able to just remove this function.

File: Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h (hunk 2 above)
Comment (VectorGamez) [reply Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h]: Done

File: Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h (hunk 1 above)
Comment (VectorGamez) [reply Source/GASExtensions/Public/GASExtAbilitySystemFunctionLibrary.h]: Done.

**Pull request merged**